packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/tasks/ScenarioExtraction.java

```diff
@@ -7,20 +7,23 @@
 import software.uncharted.terarium.hmiserver.utils.GreekDictionary;
 
 import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
 
 public class ScenarioExtraction {
 	public static List<ModelParameter> getModelParameters(JsonNode condition, Model modelCopy) {
 		final List<ModelParameter> modelParameters = modelCopy.getParameters();
 		modelParameters.forEach((parameter) -> {
 			final String parameterId = parameter.getId();
-			final JsonNode conditionParameters = condition.get("parameters");
-			conditionParameters.forEach((conditionParameter) -> {
-				// Get the parameter value from the condition
-				final String id = conditionParameter.get("id").asText();
+			condition.get("values").forEach((conditionParameter) -> {
```

Comment:
Is there ever any risk of these json nodes being not present? A
```
try {
...
} catch (exception e) {
continue;
}
```

on this and the other change may save a bunch of headaches later

Reply:
We have a try catch above that function that's handled.